n2v/methods/rmks.py

```diff
@@ -6,7 +6,521 @@
+                When this is given, the final result will be represented
+            v_tol: float, opt
+                convergence criteria for vxc Fock matrices.
+                default: 1e-4
+            D_tol: float, opt
+                convergence criteria for density matrices.
+                default: 1e-7
+            eig_tol: float, opt
+                convergence criteria for occupied eigenvalue spectrum.
+                default: 1e-4
+            frac_old: float, opt
+                Linear mixing parameter for current vxc and old vxc.
+                If 0, no old vxc is mixed in.
+                Should be in [0,1)
+                default: 0.5.
+            init: string, opt
+                Initial guess method.
+                default: "SCAN"
+                1) If None, input wfn info will be used as initial guess.
+                2) If "continue" is given, then it will not initialize
+                but use the densities and orbitals stored. Meaningly,
+                one can run a quick WY calculation as the initial
+                guess. This can also be used to user speficified
+                initial guess by setting Da, Coca, eigvec_a.
+                3) If it's not continue, it would be expecting a
+                method name string that works for psi4. A separate psi4 calculation
+                would be performed.
+
+
+        returns:
+        ----------------------
+            all are np.ndarray of shape (num_grid_points)
+            vxc, vxchole, ebarKS, ebarWF, taup_rho_WF, taup_rho_KS
+            in eqn vxc = vxchole + ebarKS - ebarWF + taup_rho_WF - taup_rho_KS.
+            Check the original paper for the definition of each component.
+    """
+        if not self.wfn.name() in ["CIWavefunction", "RHF"]:
+            raise ValueError("Currently only supports Psi4 CI wavefunction"
+                             "inputs because Psi4 CCSD wavefunction currently "
+                             "does not support two-particle density matrices.")
+
+        if self.ref != 1:
+            raise ValueError("Currently only supports Spin-Restricted "
+                             "calculations since Spin-Unrestricted CI "
+                             "is not supported by Psi4.")
+
+        if self.guide_potential_components[0] != "hartree":
+            raise ValueError("Hartree potential is necessary as the guide potential.")
+
+        Nalpha = self.nalpha
+
+        # Preparing DFT spherical grid
+        functional = psi4.driver.dft.build_superfunctional("SVWN", restricted=True)[0]
+        self.Vpot = psi4.core.VBase.build(self.basis, functional, "RV")
+        self.Vpot.initialize()
+        self.npoints_DFT = 0
+        for blk in range(self.Vpot.nblocks()):
+            self.npoints_DFT += self.Vpot.get_block(blk).x().shape[0]
+        self.Vpot.properties()[0].set_pointers(self.wfn.Da())
+
+
+        # Preparing for WF properties
+        if self.wfn.name() == "CIWavefunction":
```

Comment:
The initial part of the mRKS involves generating the `e_bar`, `tau_p`, and `vxc_hole`. I think having a function `_generate_XXXXXXX(self, wfn)` that takes the wfn and returns both `e_bar` and `tau_p` would greatly improve the readability since it would group in a few lines all the ingredients that are required for the iterative part. 

Additionally, having somewhere in the documentation the distinction between the use of a RHFwfn and CI_wfn would be great (e.g Using CI_wfn will generate {o,t}pdm to get the generalized fock matrix, whereas RHF_wfn will use the 'usual' fock matrix)

Reply:
I have thought about this and decided not to make it a separate function for this first version:
1) it is not frequently used;
2) If in a separate function, it needs either recalculate many heavy tensor multiplication or store
    data globally which should best be local.

Reply:
I am not following how you'd have to recalculate things. 
But it is fair to say that the function will probably never be used other than here, so we can keep it like this.